src/components/newcases.js

```diff
@@ -48,88 +48,52 @@ function RateOfNewCases(props) {
-            </div>
-
-            <div
-              className={`level-item ${
-                casesMap[1].cases - casesMap[2].cases < 0
-                  ? 'is-green'
-                  : 'is-cherry'
-              }`}
-            >
-              <h5 className="heading">{casesMap[1].date}</h5>
-              <h2 className="title has-text-info">{casesMap[1].cases}</h2>
-            </div>
-
-            <div className="level-item is-blue">
-              <h5 className="heading"> </h5>
-              <h1>
-                {casesMap[0].cases - casesMap[1].cases > 0
-                  ? upArrow
-                  : downArrow}
-                {Math.abs(casesMap[0].cases - casesMap[1].cases)}*
-              </h1>
-              <h1 className="title has-text-grey">{}</h1>
-            </div>
-
-            <div className="level-item is-blue">
-              <h5>{casesMap[0].date}</h5>
-              <h2> {casesMap[0].cases}* </h2>
-            </div>
+            {(function () {
+              const rows = [];
+              for (let index = casesMap.length - 1; index > 0; index--) {
+                rows.push(
+                  <React.Fragment>
+                    {index < casesMap.length - 1 && (
+                      <div
+                        className={`level-item ${
+                          casesMap[index - 1].cases - casesMap[index].cases < 0
+                            ? casesMap[index - 1].date == 'Today'
+                              ? 'is-blue'
+                              : 'is-green'
+                            : 'is-cherry'
+                        }`}
+                      >
+                        <h5 className="heading"> </h5>
+                        <h1>
+                          {casesMap[index - 1].cases - casesMap[index].cases > 0
+                            ? upArrow
+                            : downArrow}
+                          {Math.abs(
+                            casesMap[index - 1].cases - casesMap[index].cases
+                          )}
+                        </h1>
+                        <h1 className="title has-text-grey">{}</h1>
+                      </div>
+                    )}
+                    <div
+                      className={`level-item ${
+                        casesMap[index - 1].cases - casesMap[index].cases < 0
+                          ? casesMap[index - 1].date == 'Today'
+                            ? 'is-blue'
+                            : 'is-green'
+                          : 'is-cherry'
+                      }`}
+                    >
+                      <h5 className="heading">{casesMap[index - 1].date}</h5>
+                      <h2 className="title has-text-grey">
+                        {casesMap[index - 1].cases}
+                      </h2>
+                    </div>
+                  </React.Fragment>
+                );
+              }
+              return rows;
+            })()}
```

Comment:
Why function? Just move this into function body.

Reply:
created new function for this as it was hard to maintain and directly called this here.

Reply:
You don't need a function just declare the array in the function scope as just another variable.

Reply:
It looks more modular now and easy to maintain.

Reply:
It is not anymore modular. It's just more cruft.

Reply:
So should I change this ? or how can we make it better ?

Reply:
Yes. Remove the function then you can use `rows` variable directly in the return expression

Reply:
done